scripts/clang-tidy-diff.sh

```diff
@@ -27,11 +21,11 @@ if [ ! -e "$CLANG_TIDY_DIFF" ]; then
   echo "Failed to find clang-tidy-diff.py ($CLANG_TIDY_DIFF)"
   exit 1
 fi
-TIDY_MSG=$(git diff -U0 $BRANCH... | $CLANG_TIDY_DIFF -quiet -p1 2> /dev/null)
+TIDY_MSG=$(git diff -U0 $GITHUB_BASE_REF $GITHUB_HEAD_REF | $CLANG_TIDY_DIFF -quiet -p1 2> /dev/null)
```

Comment:
I think the `...` is probably still a good thing just to be precise here.   We only care about change that on the RHS here.   I know its unlikely that the BASE could have change that are not in HEAD, but in theory its possible.